File: RPC/04-2023/B.py
import math 

#0.002292
def main():

    PI = 3.1415926535897932
    n = int(input())

    points = []

    for i in range(n):
        inn = input().split()
        points.append([int(inn[0]), int(inn[1])])

    # print(points)

    start = points[0]
    end = points[n-1]

    maxi = -1
    for i in range(1,n):
        val = math.atan(abs(points[i][1])/abs(start[0] - points[i][0]))*180/PI
        maxi = max(maxi, val)
    
    for i in range(0,n-1):
        val = math.atan(abs(points[i][1])/abs(end[0] - points[i][0]))*180/PI
        maxi = max(maxi, val)

    print(maxi)

main()

File: RPC/04-2023/B.cpp
/*  

Metodo: 
Jhon Alex Gaviria
*/
#include <bits/stdc++.h>
#include <math.h>  

using namespace std; 

const int MAXDATA = 1e6 + 6;
const int oo = 1e9; 
long double PI = 3.1415926535897932384626433832795028841;

int main() {
    
    ios::sync_with_stdio(0);
	cin.tie(0);
    
    int n, m; 

    cin >> n; 

    vector<pair<long double, long double>> v; 

    for(int i=0; i<n; i++){
        int a, b; 
        cin >> a >> b; 
        v.push_back({a, b});
    }

    pair<int, int> start = v[0]; 
    pair<int, int> end = v[n-1]; 

    long double deg = 180/PI;
    long double maxi = -1; 
    for(int i=1; i<n; i++){
        // long double div = ;
        long double value = atan(abs(start.second - v[i].second)/abs(start.first - v[i].first))*deg;
        maxi = max(maxi, value);
    }
    for(int i=0; i<n-1; i++){
        
        long double value = atan(abs(end.second - v[i].second)/abs(end.first - v[i].first))*deg;
        maxi = max(maxi, value);
    }
    cout << fixed << setprecision(10) << maxi << endl;
}